provisioning_templates/snippet/redhat_register.erb

```diff
@@ -123,55 +201,21 @@ snippet: true
-    <% ( enabled_repos = "yum-config-manager --enable #{host_param('subscription_manager_repos').gsub(',', ' ')}") if host_param('subscription_manager_repos') %>
-  <% else %>
-    <% ( enabled_repos = "subscription-manager repos --enable #{host_param('subscription_manager_repos').gsub(',', ' --enable')}") if host_param('subscription_manager_repos') %>
-  <% end %>
-  <% if host_param('http-proxy') %>
-    subscription-manager config --server.proxy_hostname='<%= host_param("http-proxy") %>'
-    <% if host_param('http-proxy-user') %>
-      subscription-manager config --server.proxy_user='<%= host_param("http-proxy-user") %>'
-    <% end %>
-    <% if host_param('http-proxy-password') %>
-      subscription-manager config --server.proxy_password='<%= host_param("http-proxy-password") %>'
-    <% end %>
-    <% if host_param('http-proxy-port') %>
-      subscription-manager config --server.proxy_port="<%= host_param('http-proxy-port') %>"
-    <% end %>
-  <% end %>
-  <% if host_param('subscription_manager_username') && host_param('subscription_manager_password') %>
-    <% if host_param('subscription_manager_pool') %>
-      subscription-manager register --name='<%= @host.name %>' --username='<%= host_param("subscription_manager_username") %>' --password='<%= host_param("subscription_manager_password") %>'
-      subscription-manager attach --pool="<%= host_param('subscription_manager_pool') %>"
-    <% else %>
-      subscription-manager register --name='<%= @host.name %>' --username='<%= host_param("subscription_manager_username") %>' --password='<%= host_param("subscription_manager_password") %>' --auto-attach
-    <% end %>
-    # workaround for RHEL 6.4 bug https://bugzilla.redhat.com/show_bug.cgi?id=1008016
-    subscription-manager repos --list > /dev/null
-    <%= enabled_repos if enabled_repos %>
-  <% elsif host_param('activation_key') %>
-    <% if host_param('subscription_manager_certpkg_url') %>
-      rpm -Uvh <%= host_param('subscription_manager_certpkg_url') %>
-    <% elsif host_param('subscription_manager_host') %>
-      rpm -Uvh <%= host_param('subscription_manager_host') %>/pub/candlepin-cert-consumer-latest.noarch.rpm
+
+    <% if host_param_true?('redhat_install_agent') %>
+    if [ -f /usr/bin/dnf ]; then
+      dnf -y install osad
+    else
+      yum -t -y install osad
+    fi
     <% end %>
-    subscription-manager register --name="<%= @host.name %>" --org="<%= host_param('subscription_manager_org') %>" --activationkey="<%= host_param('activation_key') %>"
-    # workaround for RHEL 6.4 bug https://bugzilla.redhat.com/show_bug.cgi?id=1008016
-    subscription-manager repos --list > /dev/null
-    <%= enabled_repos if enabled_repos %>
   <% else %>
-    # Not registering host.params['activation_key'] not found.
+    echo "No activation key found: Not registering"
   <% end %>
 <% end %>
+
+<% if registration_type == 'none' %>
+# No parameters found for Spacewalk or Subscription manager registration.
+<% end %>
```

Comment:
LGTM I haven't tested tho. This block seems useless to me, it will render the comment for non-RH users, I'd rather see the template to render blank when no subscription support is available. Thanks!

Reply:
Sure good point.  It's updated, there's no output from the snippet if the host won't be registered